spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java

```diff
@@ -28,7 +29,7 @@
  * @author João André Martins
  */
 @ConfigurationProperties("spring.cloud.gcp.config")
-public class GcpConfigProperties {
+public class GcpConfigProperties implements CredentialsPropertiesProvider {
```

Comment:
I would call `CredentialsSupplier`. We don't provide any properties here.

Although that is just IMO